kevm-pyk/src/kevm_pyk/kproj/evm-semantics/lemmas/bytes-simplification.k

```diff
@@ -59,11 +59,17 @@ module BYTES-SIMPLIFICATION [symbolic]
       requires lengthBytes(B) <=Int 32
        andBool #rangeUInt(256, X) [simplification, concrete(B), comm]
 
-    rule [buf-asWord-invert-lr]:
+    rule [buf-asWord-invert-lr-len-leq]:
       #buf (W:Int , #asWord(B:Bytes)) => #buf(W -Int lengthBytes(B), 0) +Bytes B
       requires lengthBytes(B) <=Int W andBool W <=Int 32
       [simplification]
 
+    rule [buf-asWord-invert-lr-len-gt]:
+      #buf (W:Int , #asWord(B:Bytes)) => #range(B, lengthBytes(B) -Int W, W)
+      requires W <Int lengthBytes(B) andBool lengthBytes(B) <=Int 32
+       andBool #asWord(B) <Int 2 ^Int (8 *Int W)
+      [simplification]
```

Comment:
This simplification covers the case brought up by @lucasmt in [this comment](https://github.com/runtimeverification/evm-semantics/pull/2307/files#r1498050387).